test_unstructured_ingest/test-ingest.sh

```diff
@@ -45,29 +45,46 @@ scripts=(
 'test-ingest-sharepoint.sh'
 )
 
-CURRENT_SCRIPT="none"
+full_python_matrix_tests=(
+  'test-ingest-sharepoint.sh'
+  'test-ingest-local.sh'
+  'test-ingest-local-single-file.sh'
+  'test-ingest-local-single-file-with-encoding.sh'
+  'test-ingest-local-single-file-with-pdf-infer-table-structure.sh'
+  'test-ingest-s3.sh'
+  'test-ingest-google-drive.sh'
+  'test-ingest-gcs.sh'
+)
+
+CURRENT_TEST="none"
 
 function print_last_run() {
-  if [ "$CURRENT_SCRIPT" != "none" ]; then
-    echo "Last ran script: $CURRENT_SCRIPT"
+  if [ "$CURRENT_TEST" != "none" ]; then
+    echo "Last ran script: $CURRENT_TEST"
   fi
 }
 
 trap print_last_run EXIT
 
-for script in "${scripts[@]}"; do
-  CURRENT_SCRIPT=$script
-  if [[ "$CURRENT_SCRIPT" == "test-ingest-notion.sh" ]]; then
-    echo "--------- RUNNING SCRIPT $script --- IGNORING FAILURES"
+python_version=$(python --version 2>&1)
+
+for test in "${all_tests[@]}"; do
```

Comment:
can we add some documentation here on what this logic is meant to be handling? It's getting pretty complicated and reading bash logic comparisons are not the easiest.